src/main.rs

```diff
@@ -1,47 +1,216 @@
+#![allow(clippy::unusual_byte_groupings)]
+#![allow(clippy::upper_case_acronyms)]
```

Comment:
This will be our little secret ;)

Reply:
The 1st one I think it's better to use the same bit patterns as in the specification.

The 2nd one is to use the same acronyms as in the specification as well,t hough can be changed.